### ISSUE TITLE
IE10 triggers onChange twice when replacing a text selection 

### ISSUE THREAD

**Issue by jfschwarz** [facebook/react#8531]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
- Type text "123" in an input
- Select the whole text
- Type the `A` key

When hitting the `A` key, `onChange` is triggered twice. One time `event.target.value` is `""`. The second time it's `"A"`.

This only happens in IE10. IE9 and IE11 both behave correctly.

**Demo fiddle**

https://jsfiddle.net/69z2wepo/64637/

![ie10onchange](https://cloud.githubusercontent.com/assets/524089/21014764/8883c6fa-bd5f-11e6-8e9c-65a957bbca7e.gif)

**What is the expected behavior?**
When hitting the `A` key, `onChange` is triggered only one time. `event.target.value` should be `"A"`.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

- React 15.4.1 (possibly also older versions)
- IE10 / Windows 7

**Comment by coreyleelarson** [comment]
I am currently running into this issue as well. Someone please help!

**Comment by aweary** [comment]
Is this still an issue with React 16?

**Comment by suhailnaw** [comment]
I am attempting to look into this, but need some help! @aweary 

What's the best practices for debugging older versions of IE?  I'm currently running a VM on my Mac with Parallels and have IE11 up.  However, JSFiddle won't load the React debugging starter code (https://jsfiddle.net/Luktwrdm) in IE11.  I keep on getting "SCRIPT002: Syntax error" on the first line of the html, which is <!DOCTYPE html>.  I think it's because of the Babel transpiling not working on IE11 on JSFiddle...  Any tips?

<img width="870" alt="screen shot 2018-03-25 at 2 35 48 pm" src="https://user-images.githubusercontent.com/26286035/37880776-6855ff40-3042-11e8-9767-1ba518362f76.png">

**Comment by suhailnaw** [comment]
I tried another approach where I put the bug in an empty React app made with create-react-app, ran the server on my Mac, and logged onto it through my Parallels VM in IE11.  I used IE11's emulation feature to emulate IE10 and couldn't replicate the bug for React 16 or 15.4.1.  

I will investigate further on actual IE10 when I have access to more disk space for all these VMs, but so far no luck replicating the bug.
